packages/typespec-client-generator-core/src/package.ts

```diff
@@ -660,7 +660,18 @@ function addDefaultClientParameters<
   if (apiVersionParam) {
     client.initialization.properties.push(apiVersionParam);
   }
-  const subId = client.initialization.properties.find((x) => isSubscriptionId(context, x));
+  let subId = context.__clientToParameters
+    .get(client.__raw.type)
+    ?.find((x) => isSubscriptionId(context, x));
```

Comment:
I assume this part of the code checks whether "subscriptionId" is already added to client parameters?
Or, is it checks the "subscriptionId" on operations that does not be included in any operation group?

Reply:
Since we will update the logic in the future, I just keep same with api-verison: https://github.com/Azure/typespec-azure/blob/85d0f982f16cfb6ac44a85c989db17a7ae948064/packages/typespec-client-generator-core/src/package.ts#L647.